src/EventTrace.cc

```diff
@@ -871,63 +899,28 @@ const std::string& ValTraceMgr::ValName(const ValPtr& v)
 	auto find = val_names.find(v.get());
 	if ( find == val_names.end() )
 		{
-		if ( IsAggr(v->GetType()) )
-			{ // Aggregate shouldn't exist; create it
-			ASSERT(val_map.count(v.get()) == 0);
-			NewVal(v);
-			find = val_names.find(v.get());
-			}
-
-		else
-			{ // Non-aggregate can be expressed using a constant
-			auto tag = v->GetType()->Tag();
-			std::string rep;
-
-			if ( tag == TYPE_STRING )
-				{
-				auto s = v->AsStringVal();
-				rep = escape_string(s->Bytes(), s->Len());
-				}
-
-			else if ( tag == TYPE_LIST )
-				{
-				auto lv = cast_intrusive<ListVal>(v);
-				for ( auto& v_i : lv->Vals() )
-					{
-					if ( ! rep.empty() )
-						rep += ", ";
-
-					rep += ValName(v_i);
-					}
-				}
-
-			else if ( tag == TYPE_FUNC )
-				rep = v->AsFunc()->Name();
-
-			else if ( tag == TYPE_TIME )
-				rep = std::string("double_to_time(") + std::to_string(v->AsDouble()) + ")";
+		val_names[v.get()] = GenValName(v);
+		find = val_names.find(v.get());
```

Comment:
I'm not sure how much it matters here but if you use `insert` or `insert_or_assign` for the actual insertion, you get back a pair that includes the iterator to the element inserted. That'd avoid the extra lookup.

Reply:
Nice to know.  I've pushed this as a `--fixup`, I like how it visually streamlines the code somewhat (though also makes it a tad more cryptic due to more meta-characters).